File: simple/cppvulnweb/common/vuln_child.cpp
#include "vuln_child.h"
#include <cstdio>
#include <cstdlib>
#include <curl/curl.h>
#include <sstream>

// 回调函数
static size_t ChildWriteCallback(void* contents, size_t size, size_t nmemb, std::string* userp) {
    userp->append((char*)contents, size * nmemb);
    return size * nmemb;
}

// 子类方法实现：调用父类漏洞方法形成多层调用链
// 调用链：VulnChild -> VulnParent -> 漏洞点

// 命令执行漏洞 - 子类调用父类方法
std::string VulnChild::executeCommand(const std::string& cmd) {
    // 调用链：VulnChild -> VulnParent::executeCommand -> popen（漏洞点）
    return VulnParent::executeCommand(cmd);
}

// 代码执行漏洞 - 子类调用父类方法
std::string VulnChild::executeCode(const std::string& code) {
    // 调用链：VulnChild -> VulnParent::executeCode -> popen（漏洞点）
    return VulnParent::executeCode(code);
}

// SSRF漏洞 - 子类调用父类方法
std::string VulnChild::fetchUrl(const std::string& url) {
    // 调用链：VulnChild -> VulnParent::fetchUrl -> curl_easy_perform（漏洞点）
    return VulnParent::fetchUrl(url);
}

File: simple/kotlinvulnweb/src/main/kotlin/com/vulnweb/common/VulnChild.kt
package com.vulnweb.common

// 子类继承父类，复用漏洞方法
class VulnChild : VulnParent() {

    // 重写命令执行方法（可选择是否调用父类）
    override fun executeCommand(input: String): String {
        // 子类可以添加额外逻辑，然后调用父类漏洞方法
        val childLog = "子类日志: 准备执行命令"
        val result = super.executeCommand(input)
        return "$childLog\n$result"
    }

    // 重写代码执行方法
    override fun executeCode(input: String): String {
        val childLog = "子类日志: 准备执行代码"
        val result = super.executeCode(input)
        return "$childLog\n$result"
    }

    // 重写SSRF方法
    override fun fetchUrl(input: String): String {
        val childLog = "子类日志: 准备获取URL"
        val result = super.fetchUrl(input)
        return "$childLog\n$result"
    }
}
